### ISSUE TITLE
Sequence diagram: 

### ISSUE THREAD

**Issue by whitelizard** [mermaid-js/mermaid#1765]
### Problem
I would really like to be able to control overlapping activations. If multiple activations are added, one can not deactivate the first activation first for instance.

Consider this example, with the 2 different syntaxes:
```
sequenceDiagram
    Alice ->> + John: First question
    Alice ->> + John: Second question
    John -->> - Alice: First answer
    John -->> - Alice: Second answer
```

```
sequenceDiagram
    Alice ->> John: First question
    activate John
    Alice ->> John: Second question
    activate John
    John -->> Alice: First answer
    deactivate John
    John -->> Alice: Second answer
    deactivate John
```

The first answer should be for the first question, etc. But this can not be achieved as I understand it.

### Solution
Ability in the syntax to have a label/id for an activation would be great, and solve this matter nicely.

Example extended syntax for both cases: 
```
sequenceDiagram
    Alice ->> John: First question
    activate John: q1
    Alice ->> John: Second question
    activate John: q2
    John -->> Alice: First answer
    deactivate John: q1
    John -->> Alice: Second answer
    deactivate John: q2
```

```
sequenceDiagram
    Alice ->> +(q1) John: First question
    Alice ->> +(q2) John: Second question
    John -->> -(q1) Alice: First answer
    John -->> -(q2) Alice: Second answer
```

### Alternatives

Or, for the explicit syntax (`activate` keyword), and to be consistent, it might be better with this syntax:
```
sequenceDiagram
    Alice ->> John: First question
    activate(q1) John
    Alice ->> John: Second question
    activate(q2) John
    John -->> Alice: First answer
    deactivate(q1) John
    John -->> Alice: Second answer
    deactivate(q2) John
```

This is in the end the look I'm after:
![image](https://user-images.githubusercontent.com/8565497/98677307-2125fb00-235d-11eb-95c6-b54cb4a62941.png)

**Comment by nprezive** [comment]
Any update on this? I would love to see this feature.

**Comment by rocketraman** [comment]
I think related to this is the ability to chain activations using a shortcut syntax e.g. to get this:

![image](https://user-images.githubusercontent.com/53049/168913259-d81be2f3-bd8e-4c2d-86a0-55eccdb5b41b.png)

I have to use the extended syntax to activate the chained activation of `Bob`, since there is no way to both deactivate and activate at line 2.

```
sequenceDiagram
    Alice ->> + John: First question
    John ->> - Bob: Second question
    activate Bob
    Bob ->> - Ron: Third question
```

With @whitelizard 's label approach, this could be cleaner:

```
sequenceDiagram
    Alice ->> +(q1) John: First question
    John ->> -(q1) +(q2) Bob: Second question
    Bob ->> -(q2) Ron: Third question
```

although for this particular case I'd prefer a simpler solution that doesn't require labels by just allowing both the deactivation and activation shortcuts together:

```
sequenceDiagram
    Alice ->> + John: First question
    John ->> - + Bob: Second question
    Bob ->> - Ron: Third question
```

**Comment by JasonArm** [comment]
Agreed @whitelizard has a point - Sequence diagrams assume a LIFO order to stacked Activations.  But often queue (FIFO) or parallel activity durations need to be shown. Labelled activations would help - although a more complicated syntax {but I can't suggest anything less complex :) }
It would be great to have this feature - as it is quite a big limitation

**Comment by JasonArm** [comment]
I also agree with @rocketraman - This is a slightly different use case though, but one with a simpler syntax solution.
Activation & Deactivation in the same message makes sense.
Certainly, given that:
- Activation  changes the state of the 2nd Actor listed 
       e.g. John in this case, ` Alice ->> + **John**: First question`
- Deactivation changes the state of the 1st Actor listed
       e.g. Bob in this case `Bob ->> - Ron: Third question`
So having `- +` in a message definition is not a conflict (Active/Deactivate of the same Actor at the same time)
And, there is no ambiguity as to which Actor is activated and which is deactivated.

**Comment by mansona** [comment]
I've just come across this same issue and I would probably go for a simpler approach. If you want to have some other order than LIFO you can't use the shorthand for activations. This way you can just add a syntax for labeling activations with the `activate Actor` line, and then your activations and deactivations will need to match. e.g. (taken from @whitelizard 's example with a very slight change in syntax)

```
sequenceDiagram
    Alice ->> John: First question
    activate John q1
    Alice ->> John: Second question
    activate John q2
    John -->> Alice: First answer
    deactivate John q1
    John -->> Alice: Second answer
    deactivate John q2
```

**Comment by mansona** [comment]
if we can agree on the syntax then I could try an implementation 😂

**Comment by pseidemann** [comment]
what do you think about this one?

explicit form

```
activate John as q1
deactivate John as q1
```

shorthand form

```
Alice ->> +[q1] John: First question
John -->> -[q1] Alice: First answer
```
